BH TTNN Unit failure in test for gather

https://github.com/tenstorrent/tt-metal/actions/runs/15044479668/job/42283886992
```
FAILED tests/ttnn/unit_tests/operations/test_gather.py::test_gather_general[input_shape=[8, 8, 8, 8]-index_shape=[8, 8, 8, 8]-dim=-1] - RuntimeError: TT_FATAL @ /work/tt_metal/common/core_coord.cpp:47: end_coord.x >= start_coord.x and end_coord.y >= start_coord.y
info:
Invalid core range for start_coord: (x=0,y=4), end_coord: (x=11,y=3)
```

--- a/tests/ttnn/unit_tests/operations/reduce/test_gather.py
+++ b/tests/ttnn/unit_tests/operations/reduce/test_gather.py
@@ -84,7 +84,6 @@ def test_gather_preallocated_output(input_shape, index_shape, dim, device):
     assert_with_pcc(torch_gather, ttnn.to_torch(ttnn_output))
 
 
-# @skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
 @pytest.mark.parametrize(
     "input_shape, index_shape, dim",
     [

--- a/tests/ttnn/unit_tests/operations/reduce/test_gather.py
+++ b/tests/ttnn/unit_tests/operations/reduce/test_gather.py
@@ -9,7 +9,6 @@ from models.utility_functions import skip_for_blackhole
 from tests.ttnn.utils_for_testing import assert_with_pcc
 
 
-@skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
 @pytest.mark.parametrize(
     "input_shape, index_shape, dim",
     [
@@ -54,7 +53,6 @@ def test_gather_general(input_shape, index_shape, dim, device):
     assert_with_pcc(torch_gather, ttnn.to_torch(ttnn_gather))
 
 
-@skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
 @pytest.mark.parametrize(
     "input_shape, index_shape, dim",
     [
@@ -86,7 +84,7 @@ def test_gather_preallocated_output(input_shape, index_shape, dim, device):
     assert_with_pcc(torch_gather, ttnn.to_torch(ttnn_output))
 
 
-@skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
+# @skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
 @pytest.mark.parametrize(
     "input_shape, index_shape, dim",
     [
@@ -115,7 +113,6 @@ def test_gather_multicore_cases(input_shape, index_shape, dim, device):
     assert_with_pcc(torch_gather, ttnn.to_torch(ttnn_gather))
 
 
-@skip_for_blackhole("Gather needs to be tested and is failing on BH. Issue #22147")
 @pytest.mark.parametrize(
     "input_shape, index_shape, dim, torch_input_datatype, ttnn_input_datatype, ttnn_index_datatype",
     [

--- a/ttnn/cpp/ttnn/operations/experimental/gather/device/gather_program_factory.cpp
+++ b/ttnn/cpp/ttnn/operations/experimental/gather/device/gather_program_factory.cpp
@@ -88,7 +88,7 @@ GatherProgramFactory::cached_program_t GatherProgramFactory::create(
             if (core_grid_calculated_columns_number != 0) {
                 const CoreRange additional_range(
                     {0, core_grid_calculated_rows_number},
-                    {core_grid_calculated_columns_number - 1, core_grid_calculated_rows_number - 1});
+                    {core_grid_calculated_columns_number, core_grid_calculated_rows_number});
                 core_range = core_range.merge(CoreRangeSet(additional_range));
             }
         }

--- a/ttnn/cpp/ttnn/operations/experimental/gather/gather.cpp
+++ b/ttnn/cpp/ttnn/operations/experimental/gather/gather.cpp
@@ -51,7 +51,7 @@ Tensor pre_gather_transform_tensor(
         index_tensor_padded_shape[0],
         index_tensor_padded_shape[1],
         index_tensor_padded_shape[2],
-        input_tensor.get_logical_shape()[-1]};
+        transformed_tensor.get_logical_shape()[-1]};
 
     const Tensor sliced_tensor =
         ttnn::slice(transformed_tensor, start_index, end_index, step, input_tensor.memory_config());

